### PR TITLE
Import SAME70 DFP version 2.4.166

### PR DESCRIPTION
**Pull request by stephanosio**
Import SAME70 DFP version 2.4.166 for SAM E70 and SAM E70B.

NOTE: The "PATCH: Fix GMAC priority queue register offsets" was not applied to `same70b` because this is already fixed in the upstream.

Supersedes #7.

### CODE REVIEW

**Comment by stephanosio** [comment]
@nashif @galak "Scancode" fails due to the missing copyright notice and license identifier in some files. Any suggestion on how to resolve this without modifying the affected files (e.g. adding license information in the commit message instead)? These files are directly from Microchip and I would prefer not to touch them; the entire DFP package is Apache 2.0 licensed, so no real issue here)?

**Comment by stephanosio** [comment]
cc @nandojve

**Comment by stephanosio** [comment]
@nashif @galak Could you review/merge this?

File: asf/sam/include/same70/component/xdmac.h [hunk 1]
```diff
@@ -2396,8 +2460,8 @@ typedef struct {
   __O  uint32_t XDMAC_GSWR;     /**< (XDMAC Offset: 0x38) Global Channel Software Request Register */
   __I  uint32_t XDMAC_GSWS;     /**< (XDMAC Offset: 0x3C) Global Channel Software Request Status Register */
   __O  uint32_t XDMAC_GSWF;     /**< (XDMAC Offset: 0x40) Global Channel Software Flush Request Register */
-  RoReg8  Reserved1[0x0C];
-       XdmacChid XDMAC_CHID[XDMACCHID_NUMBER]; /**< Offset: 0x50 Channel Interrupt Enable Register (chid = 0) */
+  __I  uint8_t                        Reserved1[12];
+       XdmacChid XdmacChid[XDMACCHID_NUMBER]; /**< Offset: 0x50 Channel Interrupt Enable Register (chid = 0) */
```
Comment (mnkp) [inline asf/sam/include/same70/component/xdmac.h]: In the old implementation this field was called `XDMAC_CHID`, now we have `XdmacChid`. This is causing a compile time error in `drivers/dma/dma_sam_xdmac.c`.

File: asf/sam/include/same70/component/tc.h [hunk 2]
```diff
@@ -939,21 +903,21 @@ typedef struct {
...
   __IO uint32_t TC_EMR;         /**< (TC_CHANNEL Offset: 0x30) Extended Mode Register (channel = 0) */
-       RoReg8                         Reserved1[0x0C];
+  __I  uint8_t                        Reserved1[12];
 } TcChannel;
 
 #define TCCHANNEL_NUMBER 3
 /** \brief TC hardware registers */
 typedef struct {  
-       TcChannel TC_CHANNEL[TCCHANNEL_NUMBER]; /**< Offset: 0x00 Channel Control Register (channel = 0) */
+       TcChannel TcChannel[TCCHANNEL_NUMBER]; /**< Offset: 0x00 Channel Control Register (channel = 0) */
```
Comment (mnkp) [inline asf/sam/include/same70/component/tc.h]: Same story like with the `XDMAC_CHID`.

File: asf/sam/include/same70/component/usbhs.h [hunk 3]
```diff
@@ -3299,23 +3415,23 @@ typedef struct {
...
+  __I  uint8_t                        Reserved6[8];
   __O  uint32_t USBHS_DEVEPTIER[10]; /**< (USBHS Offset: 0x1F0) Device Endpoint Enable Register (n = 0) 0 */
-  RoReg8  Reserved7[0x8];
+  __I  uint8_t                        Reserved7[8];
   __O  uint32_t USBHS_DEVEPTIDR[10]; /**< (USBHS Offset: 0x220) Device Endpoint Disable Register (n = 0) 0 */
-  RoReg8  Reserved8[0xC8];
-       UsbhsDevdma USBHS_DEVDMA[USBHSDEVDMA_NUMBER]; /**< Offset: 0x310 Device DMA Channel Next Descriptor Address Register (n = 1) */
-  RoReg8  Reserved9[0x80];
+  __I  uint8_t                        Reserved8[200];
+       UsbhsDevdma UsbhsDevdma[USBHSDEVDMA_NUMBER]; /**< Offset: 0x310 Device DMA Channel Next Descriptor Address Register (n = 1) */
```
Comment (mnkp) [inline asf/sam/include/same70/component/usbhs.h]: Again, name change.

File: asf/sam/include/same70/component/xdmac.h (hunk 1 above)
Comment (stephanosio) [reply asf/sam/include/same70/component/xdmac.h]: Thanks for reviewing.

> In the old implementation this field was called `XDMAC_CHID`, now we have `XdmacChid`. This is causing a compile time error in `drivers/dma/dma_sam_xdmac.c`.

Ok, I will add a patch to fix that name.

Comment (mnkp) [reply asf/sam/include/same70/component/xdmac.h]: I didn't go through all the files, there may be more places that need fixing.

**Comment by stephanosio** [comment]
@carlescufi Could you merge this for https://github.com/zephyrproject-rtos/zephyr/pull/23055?

**Comment by galak** [comment]
Reviewed scancode failures and ok w/them.

**Pull request merged**